internal/doc/invoice.go

```diff
@@ -155,6 +178,37 @@ func newClaves(inv *bill.Invoice) []IDClave {
 	return claves
 }
 
+func newFacturaRectificativa(inv *bill.Invoice) *FacturaRectificativa {
+	if len(inv.Preceding) == 0 {
+		return nil
+	}
+
+	p := inv.Preceding[0]
+
+	return &FacturaRectificativa{
+		Codigo: p.Ext[es.ExtKeyTBAICorrection].String(),
+		Tipo:   CorrectiveTypeDifferences, // Only differences are supported for now
+	}
+}
+
+func newFacturasRectificadasSustituidas(inv *bill.Invoice) *FacturasRectificadasSustituidas {
+	if inv.Preceding == nil {
+		return nil
+	}
+
+	p := inv.Preceding[0]
+
+	return &FacturasRectificadasSustituidas{
+		IDFacturaRectificadaSustituida: []*IDFacturaRectificadaSustituida{
+			{
+				SerieFactura:           p.Series,
+				NumFactura:             p.Code,
+				FechaExpedicionFactura: formatDate(p.IssueDate),
```

Comment:
I think this is fine. I honestly would not try and extract the original documents date from the signature unless this actually causes an error.

Reply:
I checked that no validation is performed at the time of posting. You can actually put any value as the series, num and fecha expedición of the reference invoice. I suppose they do it like this to allow to send a credit note for an invoice that hasn't been posted yet. TBAI seems to have a quite liberal implementation in this regard.